pkg/applicationserver/io/packages/loragls/v3/package.go

```diff
@@ -136,7 +178,60 @@ func minInt(a int, b int) int {
 	return b
 }
 
-func (p *GeolocationPackage) multiFrameQuery(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers, up *ttnpb.ApplicationUplink, data *Data, client *api.Client) (api.AbstractLocationSolverResponse, error) {
+// pushUplink updates the package data with the given uplink.
+func (p *GeolocationPackage) pushUplink(
+	ctx context.Context,
+	ids *ttnpb.ApplicationPackageAssociationIdentifiers,
+	up *ttnpb.ApplicationUplink,
+	data *Data,
+) (*ttnpb.ApplicationPackageAssociation, error) {
+	maxUplinkCount := minInt(data.MultiFrameWindowSize, 16)
```

Comment:
I think zero should mean that the default is used too. I don't think we should allow infinite storage of metadata.

Reply:
Should I remove the `maxUplinkCount != 0` check?

Reply:
I would say that a `maxUplinkCount` of zero after this clamping (`minInt`) should turn `maxUplinkCount` into 16.